server/app/repository/VersionRepository.java

```diff
@@ -87,11 +90,12 @@ private enum PublishMode {
   }
 
   private Version publishNewSynchronizedVersion(PublishMode publishMode) {
+    // Regardless of whether changes are published or not, we still perform
+    // this operation inside of a transaction in order to ensure we have
+    // consistent reads.
+    Transaction transaction =
+        database.beginTransaction(TxScope.requiresNew().setIsolation(TxIsolation.SERIALIZABLE));
```

Comment:
**question (non-blocking):** why move this out of the `try` block?

Reply:
I saw some inconsistencies with sometimes using `database.commitTransaction();` / `database.endTransaction();` and other times defining a transaction and using `transaction.commit();` /  `transaction.end();`. Thought I'd make this consistent by using the transaction throughout the function, but in order to use it for transaction.end() it needed to be defined outside of the block.